File: C/Lv.0/14.c
//피자 나눠먹기(1)
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int solution(int n) {
    int answer = 0;
    int tmp = 0;
    
    tmp = n % 7;
    
    if (tmp > 0)
        answer = n / 7 + 1;
    else
        answer = n / 7;
    
    return answer;
}

File: Python/Lv.2/14.py
//[1차] 캐시
def solution(cacheSize, cities):
    answer = 0
    tmp = [];
    tr = 0;
    ti = 0;
    
    
    
    if(cacheSize == 0):
        return len(cities) * 5;
    
    for i in cities:
        i = i.upper();
        if(len(tmp) < cacheSize and i not in tmp):
            tmp.insert(0, i);
            answer += 5;
        elif(i in tmp):
            tr = tmp.index(i);
            ti = tmp[tr];
            tmp[1:tr + 1] = tmp[0 : tr];
            tmp[0] = ti;
            answer += 1;
        elif(len(tmp) == cacheSize):
            tmp[1 : cacheSize] = tmp[0 : cacheSize - 1];
            tmp[0] = i;
            answer += 5;
    
    return answer
